src/video/scalebit.c

```diff
@@ -50,7 +50,7 @@
 static inline void stage_scale2x(void* dst0, void* dst1, const void* src0, const void* src1, const void* src2, unsigned pixel, unsigned pixel_per_row)
 {
 	switch (pixel) {
-#if defined(__x86_64__) || defined(_M_X64) || defined(__SSE2__) || (defined(_M_IX86_FP) && (_M_IX86_FP == 2))
+#if defined(__x86_64__) || defined(_M_X64) || defined(__SSE2__) || (defined(_M_IX86_FP) && (_M_IX86_FP == 2) && defined(WITH_SSE2))
```

Comment:
Hey, I found a really interesting topic:
https://social.msdn.microsoft.com/Forums/vstudio/en-US/e9ee12d7-6d6e-4f80-90a4-35b08f383c3f/illegal-code-generated-by-vc2013-compiler-x64-or-x86-with-sse-instructions?forum=vcgeneral
If you surround scale2x_8_sse2_border(...) with #pragma optimize("", off) / #pragma optimize("", on) problem is fixed.
**It seems a real serious bug of MSVC 12 optimizer in Release compiling mode**.
If you don't like WITH_SSE2 option, we could do so:
```
#if defined(_MSC_VER) && _MSC_VER == 1800
#pragma optimize("", off)
#endif
static inline void scale2x_8_sse2_border(scale2x_uint8* dst, const scale2x_uint8* src0, const scale2x_uint8* src1, const scale2x_uint8* src2, unsigned count)
{
[...]
}
#if defined(_MSC_VER) && _MSC_VER == 1800
#pragma optimize("", on)
#endif
```